apps/user-ms/src/main/java/it/pagopa/selfcare/user/controller/UserController.java

```diff
@@ -77,6 +80,28 @@ public Uni<UserProductsResponse> getUserProductsInfo(@PathParam(value = "userId"
                 .map(userMapper::toUserProductsResponse);
     }
 
+    /**
+     * getUserById function returns all the information of a user retrieved from pdv given the userId
+     * @param userId String
+     * @return A uni UserDetailResponse
+     */
+    @Operation(summary = "Retrieves user's information from pdv")
+    @GET
+    @Path("/{id}/details")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Uni<UserDetailResponse> getUserDetailsById(@PathParam(value = "id") String userId) {
+        return userService.getUserById(userId).onItem().transform(userMapper::toUserResponse);
+    }
+
+    @Operation(summary = "Search user by fiscalCode")
+    @POST
+    @Path("/search")
```

Comment:
you could use GET method with query params for search with few params, path could be "/"

Reply:
Even tho the parameter in question is the user's fiscalCode, which is classified as sensitive data?